File: Back-end/user-service/src/main/java/org/sid/userservice/service/SkillService.java
package org.sid.userservice.service;


import org.sid.userservice.entity.Skill;

import java.util.Collection;

public interface SkillService {

    void saveSkill(String skillName);
    void addSkillToUser(String skillName);
    Collection<Skill> getCurrentUserSkills();
    Collection<Skill> getUserSkills(Long id);
}

File: Front-end/src/app/user/shared/skill/skill.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams} from '@angular/common/http';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class SkillService {

  url : string = "http://localhost:8888/USER-SERVICE/api/skill";
  url2 : string = "http://localhost:8081/api/skill";

  constructor(private http : HttpClient) { }

  addUserSkill(skillName : string){
    let param = new HttpParams();
    param = param.append('skillName',skillName);
    return this.http.get(this.url2+"/save",{params : param})
  }

  getCurrentUserSkills():Observable<string[]>{
    return this.http.get<string[]>(this.url2);
  }

  getUserSkills(id : number):Observable<string[]>{
    return this.http.get<string[]>(this.url2+"/"+id);
  }
}
